config/externalnamenottested.go

```diff
@@ -143,4 +143,127 @@ var ExternalNameNotTestedConfigs = map[string]config.ExternalName{
+	// {{name}}
+	"google_access_context_manager_gcp_user_access_binding": config.IdentifierFromProvider,
+	// {{name}}
+	"google_access_context_manager_service_perimeter": config.NameAsIdentifier,
+	// {{perimeter_name}}/{{resource}}
+	"google_access_context_manager_service_perimeter_resource": config.TemplatedStringAsIdentifier("resource", "{{ .parameters.perimeter_name }}/{{ .external_name }}"),
+
+	// activedirectory
+	//
+	// {{name}}: projects/{project}/locations/global/domains/{domainName}
+	"google_active_directory_domain": config.TemplatedStringAsIdentifier("domain_name", "projects/{{ .setup.configuration.project }}/locations/global/domains/{{ .external_name }}"),
+	// projects/{{project}}/locations/global/domains/{{domain}}/{{target_domain_name}}
+	"google_active_directory_domain_trust": config.TemplatedStringAsIdentifier("target_domain_name", "projects/{{ .setup.configuration.project }}/locations/global/domains/{{ .parameters.domain }}/{{ .external_name }}"),
+
+	// apigee
+	//
+	// {{org_id}}/endpointAttachments/{{endpoint_attachment_id}}
+	"google_apigee_endpoint_attachment": config.TemplatedStringAsIdentifier("endpoint_attachment_id", "{{ .parameters.org_id }}/endpointAttachments/{{ .external_name }}"),
+	// {{org_id}}/envgroups/{{name}}
+	"google_apigee_envgroup": config.TemplatedStringAsIdentifier("name", "{{ .parameters.org_id }}/envgroups/{{ .external_name }}"),
+	// {{envgroup_id}}/attachments/{{name}}
+	"google_apigee_envgroup_attachment": config.IdentifierFromProvider,
+	// {{org_id}}/environments/{{name}}
+	"google_apigee_environment": config.TemplatedStringAsIdentifier("name", "{{ .parameters.org_id }}/environments/{{ .external_name }}"),
+	// {{org_id}}/environments/{{environment}} roles/viewer
+	"google_apigee_environment_iam_binding": config.TemplatedStringAsIdentifier("", "{{ .parameters.org_id }}/environments/{{ .parameters.env_id }} {{ .parameters.role }}"),
+	// {{org_id}}/environments/{{environment}} roles/viewer user:jane@example.com
+	"google_apigee_environment_iam_member": config.TemplatedStringAsIdentifier("", "{{ .parameters.org_id }}/environments/{{ .parameters.env_id }} {{ .parameters.role }} {{ .parameters.member }}"),
+	// {{org_id}}/environments/{{environment}}
+	"google_apigee_environment_iam_policy": config.TemplatedStringAsIdentifier("", "{{ .parameters.org_id }}/environments/{{ .parameters.env_id }}"),
+	// {{org_id}}/instances/{{name}}
+	"google_apigee_instance": config.TemplatedStringAsIdentifier("name", "{{ .parameters.org_id }}/instances/{{ .external_name }}"),
+	// {{instance_id}}/attachments/{{name}}
+	"google_apigee_instance_attachment": config.IdentifierFromProvider,
+	// organizations/{{name}}
+	"google_apigee_organization": config.IdentifierFromProvider,
+
+	// apikeys
+	//
+	// projects/{{project}}/locations/global/keys/{{name}}
+	"google_apikeys_key": config.TemplatedStringAsIdentifier("name", "projects/{{ .setup.configuration.project }}/locations/global/keys/{{ .external_name }}"),
```

Comment:
As we have discussed offline, we had better open a new issue describing some options we have for handling the optional `project` argument and pointers to the resources where we already kept the optional `project` attribute but configured the external name so that the project is referenced from the `ProviderConfig`:
- Having specialized external-name configurations for provider-gcp which first check `spec.forProvider.project` and if empty, use the `ProviderConfig` project.
- Extend the API exposed by upjet so that we can default empty template values
- Remove the optional `project` argument from the schema so that we always use the `ProviderConfig` project in those cases (becomes hard to specify a different `project` for resources then, but at least we will not be allowing inconsistencies where a `spec.forProvider.project` is specified and we are extracting the `project` from `ProviderConfig` only). And if we want to add an optional `spec.forProvider.project` later, it's a non-breaking change for the new resources.
- ...

We also need to be careful about the already configured resources this way: If we just remove the optional `project` field, it will be a breaking change that needs special handling. So I think we had better support constructing ID by first looking at `spec.forProvider.project` if it's non-empty and if it's empty, using the `ProviderConfig`'s project.